### ISSUE TITLE
ExecuteAsync sometimes broken (followup to #24)

### ISSUE THREAD

**Issue by lhaus** [egramtel/tdsharp#28]
this is still happening for me with 1.5.1 .dylib on macOS 10.15.1, just gets stuck after async execution. 
```
client.ExecuteAsync(new TdApi.GetUser
            {
                UserId = uid
            }).Result;
```

@x2bool

_Originally posted by @lhaus in https://github.com/egramtel/tdsharp/issues/24#issuecomment-561822044_

**Comment by lhaus** [comment]
can reproduce on arch linux (latest kernel + latest tdlib), and with tdlib 1.4.0 (reverted 1.5 commit)... no idea what's going on here

**Comment by lhaus** [comment]
i found out what's happening. if you send a request to tdlib (for example, getUser), while in the event handler for another event (for example, UpdateNewMessage), it hangs indefinitely... is this intentional behaviour?

**Comment by x2bool** [comment]
This is an interesting question. There is a thread behind the scenes that receives data from TDLib. When an update comes from tdlib the UpdateReceived event is fired on that thread, and in your case you schedule yet another invocation via ExecuteAsync, and that makes it to wait indefinitely on that thread.

This is obviously undesirable behavior for tdsharp. In theory this could be fixed by using dedicated thread for update events. I need some time to think about that before I fix it.

**Comment by x2bool** [comment]
Should be resolved by #34
